workflow/scripts/postprocess_go_enrichment.py

```diff
@@ -0,0 +1,47 @@
+import pandas as pd
+
+
+def string_to_tuple(string):
+    """Convert a string representation of a tuple to an actual tuple of integers."""
+    return tuple(map(int, string.replace('(', '').replace(')', '').replace(' ', '').split(',')))
+
+
+def calculate_enrichment(ratio_in_study_str, ratio_in_pop_str):
+    """Calculate enrichment based on ratios provided as strings."""
+    ratio_in_study = string_to_tuple(ratio_in_study_str)
+    ratio_in_pop = string_to_tuple(ratio_in_pop_str)
+    enrichment_study = ratio_in_study[0] / ratio_in_study[1]
+    enrichment_pop = ratio_in_pop[0] / ratio_in_pop[1]
+    return enrichment_study, enrichment_pop
+
+
+def sort_group(group):
+    """Sort a DataFrame group by the 'p_uncorrected' column in ascending order."""
+    return group.sort_values(by='p_uncorrected', ascending=True)
+
+
+# Load data
+df_enr = pd.read_csv(snakemake.input["enrichment"], sep='\t')
+df_sig = pd.read_csv(snakemake.input["significant_terms"], sep='\t')
+
+# Only keep data if GO term exists in both tables
+common_ids = df_sig[df_sig['GO'].isin(df_enr['GO'])]['GO']
+df_enr_filtered = df_enr[df_enr['GO'].isin(common_ids)]
+df_sig_filtered = df_sig[df_sig['GO'].isin(common_ids)]
+
+# Add study items from significant terms to dataset
+df_enr_filtered['study_items_sig_terms'] = df_enr_filtered['GO'].map(
+    df_sig_filtered.set_index('GO')['study_items'])
```

Comment:
This sounds like a job for a `join` to me, with `inner` as the mode for `how` in pandas:

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.join.html#pandas.DataFrame.join

I think this would be more succinct code and maybe even more readable.